torch/_inductor/kernel/flex/flex_decoding.py

```diff
@@ -88,35 +78,35 @@ def _use_flex_decoding(query, kv_indices, value, kernel_options, enable_gqa) ->
         not force_flex
         and not kernel_options.get("OUTPUT_MAX", False)
         and short_query_length
-        and can_use_single_block_m
         and static_batch
         and static_num_heads
         and non_zero_length
         and valid_block_mask_num_heads
         and pw_of_two
     )
     log.debug(
-        "Use flex decoding %s, force_flex_attention=%s, short_query_length=%s, can_use_single_block_m=%s, static_batch=%s, static_num_heads=%s",  # noqa: B950
+        "Use flex decoding %s, force_flex_attention=%s, short_query_length=%s, static_batch=%s, static_num_heads=%s",  # noqa: B950
         out,
         force_flex,
         short_query_length,
-        can_use_single_block_m,
         static_batch,
         static_num_heads,
     )
     return out
 
 
 @SymbolicGridFn
-def flex_decoding_grid(batch_size, kv_heads, gqa_group_size, n_keys, d_model, meta):
+def flex_decoding_grid(batch_size, kv_heads, gqa_group_size, seq_len_q, d_model, meta):
     """How is this kernel parallelized?
     We create a grid of (batch_size * kv_heads, SPLIT_KV, 1)
     Each block is responsible for iterating over blocks of keys and values calculating
     the local output for their tile of keys and values over all full length of query.
     groups of SPLIT_KV blocks then combine their output to produce the final result.
     """
 
-    return (batch_size * kv_heads, meta["SPLIT_KV"], 1)
+    BLOCK_M = meta["BLOCK_M"]
+    num_block_m = ceildiv(seq_len_q * gqa_group_size, BLOCK_M)
+    return (batch_size * kv_heads * num_block_m, meta["SPLIT_KV"], 1)
```

Comment:
nit: why dont we just put block_m into its own dim: https://github.com/pytorch/pytorch/blob/a2d11eb06972cbfa3fb0d23595e89d7585618e57/torch/_inductor/kernel/flex/flex_attention.py#L75